### ISSUE TITLE
[turborepo] Concurrency option not ignored when Parallel execution flag is set

### ISSUE THREAD

**Issue by nderscore** [vercel/turbo#4438]
### What version of Turborepo are you using?

1.8.4 - 1.8.8

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Mac

### Describe the Bug

According to the docs, [the `--concurrency` option will be ignored if the `--parallel` flag is also passed](https://github.com/vercel/turbo/blob/1bffb1dc8bd44b1334b37d114fe69f15103c3693/docs/pages/repo/docs/reference/command-line-reference.mdx?plain=1#L120).

Since turborepo v1.8.4, this is no longer the behavior. Setting the `--parallel` CLI flag does not disable the `--concurrency` setting when a task is persistent.

In a project with more than 10 packages, the following error message is presented for a task with the option `"persistent": true`:

```
ERROR  run failed: error preparing engine: Invalid persistent task configuration:
You have {N} persistent tasks but `turbo` is configured for concurrency of 10. Set --concurrency to at least {N+1}
```

### Expected Behavior

Setting `--parallel` should override the `--concurrency` option. 

This was the existing behavior in turborepo v1.8.3

### To Reproduce

Execute a `--parallel` task in a workspace with >10 packages and `"persistent": true` in the task's settings

**Comment by mmachatschek** [comment]
We also bumped into this issue after migrating turbo from v1.7 to 1.9

**Comment by gsoltis** [comment]
Looks like we weren't checking for `--parallel` before doing some validation. I'll see if we can fix that quickly: #5191
